share_android_ios/lib/share_android_ios.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/services.dart';
+
+class ShareAndroidIos {
+  static const MethodChannel _channel =
+      const MethodChannel('share_android_ios');
+
+  static Future<void> share(String message) async {
+    await _channel.invokeMethod('share', {'message': message});
```

Comment:
The call might fail—for example, if the platform doesn’t support the platform API (such as when running in a simulator), so wrap the invokeMethod call in a try-catch statement. (C) flutter docs